src/libpmem2/config.c

```diff
@@ -56,30 +71,70 @@ config_init(struct pmem2_config *cfg)
 }
 
 /*
- * pmem2_config_new -- allocates and initialize cfg structure.
+ * pmem2_config_new -- allocates and initialize cfg structure
  */
 int
 pmem2_config_new(struct pmem2_config **cfg)
 {
 	int ret;
-	*cfg = pmem2_malloc(sizeof(**cfg), &ret);
-
+	*cfg = pmem2_config_malloc(&ret);
 	if (ret)
 		return ret;
 
-	ASSERTne(cfg, NULL);
+	pmem2_config_init(*cfg);
 
-	config_init(*cfg);
 	return 0;
 }
 
 /*
- * pmem2_config_delete -- dealocate cfg structure.
+ * pmem2_config_delete -- deallocate cfg structure
  */
 int
 pmem2_config_delete(struct pmem2_config **cfg)
 {
 	Free(*cfg);
 	*cfg = NULL;
+
+	return 0;
+}
+
+/*
+ * pmem2_config_dup -- duplicates cfg structure
+ */
+int
+pmem2_config_dup(struct pmem2_config **dst, const struct pmem2_config *src)
+{
+	int ret;
+	struct pmem2_config *cfg = pmem2_config_malloc(&ret);
+	if (ret)
+		return ret;
+
+	memcpy(cfg, src, sizeof(*cfg));
+
+	ret = pmem2_config_file_dup(cfg, src);
+	if (ret)
+		goto err_file_dup;
+
+	*dst = cfg;
+
 	return 0;
+
+err_file_dup:
+	free(cfg);
+
+	return ret;
+}
+
+/*
+ * pmem2_config_dup_delete -- deallocates duplicated cfg structure
+ */
+int
+pmem2_config_dup_delete(struct pmem2_config **cfg)
+{
+	int ret = pmem2_config_file_dup_close(*cfg);
+
+	Free(*cfg);
```

Comment:
if pmem2_config_file_dup_close failed should we really continue with the destruction of config?